### ISSUE TITLE
if the input ply file does not contain closed point cloud, but just a surface which is viewed from top but also contains x y z coordiate and its normals.  how can i keep the input vertices boundary  is the same as output vertices boundary in x y z directions.

### ISSUE THREAD

**Issue by xxszkj123** [mkazhdan/PoissonRecon#153]
my command as follows: 
PoissonRecon.exe  --in testIn.ply --out testOut.ply --pointWeight 0

**why boundary  changes? what can i do to solve it, can you tell me some suggestions.**

my original clound points:  see attachments --- origin_cloudpoints.png  
input ply files:  see attachments ---  unzip the testIn.zip  
output ply files:  see attachments --- outputMesh.png

![origin_cloudpoints](https://user-images.githubusercontent.com/54398401/82015634-cfecc580-96b1-11ea-8b7d-ef43dd03f8bd.png)
![outputMesh](https://user-images.githubusercontent.com/54398401/82015642-d5e2a680-96b1-11ea-9987-ac45a305cb76.png)

[testIn.zip](https://github.com/mkazhdan/PoissonRecon/files/4632457/testIn.zip)

**Comment by mkazhdan** [comment]
I’m not sure I follow the question.

What is the problem with the boundary that you are trying to solve?

If you want the reconstructed surface to only go as far as the input point-set, you can reconstruct the surface with density information by calling:

PoissonRecon.exe --in testIn.ply --out testOut.ply --pointWeight 0 –density

and then trim off the parts in the low sampling density regions by calling:

              SurfaceTrimmer.exe  --in testOut.ply --out testOut.trim.ply --trim <trim value>

where, for your dataset, <trim value> should be set somewhere in the range [4,5].

 

-- Misha

**Comment by xxszkj123** [comment]
Yes, I want the surface boundary to only go as far as the input point-set.
follow your advice。
PoissonRecon.exe --in testIn.ply --out testOut.ply --pointWeight 0 –density
it doesn't have any effect.
and then
SurfaceTrimmer.exe  --in testOut.ply --out testOut.trim.ply --trim 5
The boundary is close to where it was before. but not the same. and I tried to increase the <--trim>
If the <--trim> setting is too large, it will cause the mesh net broke.  That's not what I expected either

I'm curious about that the point beyond the boundary，How was it added? Is there any way to avoid it instead of using the <--trim>.

**Comment by mkazhdan** [comment]
Unfortunately, there is not. The approach is implicit rather than interpolatory, so the vertices in the output are not the same as the input points.
